browser/src/Services/QuickOpen.ts

```diff
@@ -4,102 +4,144 @@
  * Manages the quick open menu
  */
 
-import { execSync } from "child_process"
-import * as path from "path"
-import * as Log from "./../Log"
+import { spawn } from "child_process"
 
 import * as glob from "glob"
-import * as flatten from "lodash/flatten"
-import * as Q from "q"
+import * as path from "path"
+import * as Log from "./../Log"
 
 import * as Config from "./../Config"
+import { NeovimEditor } from "./../Editor/NeovimEditor"
 import { INeovimInstance } from "./../neovim"
-import * as PromiseHelper from "./../PromiseHelper"
 import * as UI from "./../UI/index"
-import * as Git from "./Git"
 
 export class QuickOpen {
     private _seenItems: string[] = []
+    private _loadedItems: string[] = []
 
-    constructor(neovimInstance: INeovimInstance) {
+    // TODO If we export the icon, change from arg.icon to whatever
+    // Or have modes
+    constructor(neovimInstance: INeovimInstance, neovimEditor: NeovimEditor) {
         UI.events.on("menu-item-selected:quickOpen", (selectedItem: any) => {
             const arg = selectedItem.selectedOption
+
+            // If we are loading don't open this
+            if (arg.icon === "refresh fa-spin fa-fw") {
+                return
+            } else if (arg.icon === "info") {
+                // If we are info it means we need to open the config
+                neovimEditor.executeCommand("oni.config.openConfigJs")
+                return
+            }
+
             const fullPath = path.join(arg.detail, arg.label)
 
             this._seenItems.push(fullPath)
 
             if (!selectedItem.openInSplit) {
-                neovimInstance.command("tabnew! " + fullPath)
+                neovimInstance.command("e! " + fullPath)
             } else {
                 neovimInstance.command("vsp! " + fullPath)
             }
+
+            if (arg.icon === "chain") {
+                neovimInstance.chdir(fullPath)
+            }
+
         })
     }
 
-    public show(): void {
+    public async show() {
+        this._loadedItems.splice(0, this._loadedItems.length - 1)
+
         const config = Config.instance()
         const overriddenCommand = config.getValue("editor.quickOpen.execCommand")
-        const exclude = config.getValue("oni.exclude")
+        // const exclude = config.getValue("oni.exclude")
+
+        //  If in exec directory or home, show bookmarks to change cwd to
+        if (this._isInstallDirectoryOrHome()) {
+            const bookmarks = config.getValue("oni.bookmarks")
+            let icon = "chain"
+            const helpMessage = "No bookmarks yet! Select this to open config! (shows example)"
```

Comment:
oh I forgot to add that, I did on my other commit but I abandoned it because I started going in a different direction. 

And I will add an open folder option as well.

Also, do you want to test out my binary? I don't know how to add it to the install or where, I' would assume in bin...